data/common.yaml

```diff
@@ -20,6 +23,13 @@ mcollective::server_config:
   plugin.yaml: "/etc/puppetlabs/mcollective/generated-facts.yaml"
   registerinterval: 10
   registration: "Choria"
+  logfile: "/var/log/puppetlabs/mcollective.log"
```

Comment:
We already have `libdir` and `configdir` etc - for example this reason.  These are properties other modules need/want.  The more mundane things they will not goes into these hashes.

As we're in the process of moving more properties into this module this is exactly the right time to make the evaluation - is it a generically usable thing or not - if it is, its a property.

There's no question of 'if we keep $server_config` we have no choice, the possible values for `$server_config` is not enumerable, it's infinite.  But we do know some like the core server properties and pulling them out into class parameters is the right thing to do in cases where they are generally usable.